### PR TITLE
feat: coinbase wallet sdk v4 with SW support

### PR DESCRIPTION
**Pull request by joaquim-verges**
## PR-Codex overview
This PR introduces support for Coinbase Smart Wallet, updates dependencies, and refactors file imports.

### Detailed summary
- Added support for Coinbase Smart Wallet
- Updated dependencies to "@coinbase/wallet-sdk": "4.0.0-rc.1"
- Refactored file imports for consistency

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/create-wallet.ts`, `legacy_packages/contracts-js/package.json`, `packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts [hunk 1]
```diff
@@ -78,69 +99,41 @@ export type CoinbaseSDKWalletConnectionOptions = {
...
+
+async function initProvider(
+  options?: CreateWalletArgs<"com.coinbase.wallet">[1],
+): Promise<ProviderInterface> {
+  if (!_provider) {
+    const client = new CoinbaseWalletSDK({
+      appName: options?.appMetadata?.name || getDefaultAppMetadata().name,
+      appChainIds: options?.chains
+        ? options.chains.map((c) => c.id)
+        : undefined,
```
[the remark this answers is not in the document]
Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts:121]: it does under the hood in the lib if you pass undefined

File: packages/thirdweb/src/wallets/wallet-types.ts [hunk 2]
```diff
@@ -118,7 +121,9 @@ export type WalletCreationOptions<T extends WalletId> = T extends "smart"
   ? SmartWalletOptions
   : T extends "inApp" | "embedded"
     ? InAppWalletCreationOptions
-    : undefined;
+    : T extends "com.coinbase.wallet"
```
[the remark this answers is not in the document]
Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/wallet-types.ts:124]: yeah that's a good callout. we do have a 'type' for this, but not a constants file. wanna add one?

[the remark this answers is not in the document]
Comment (gregfromstl) [reply packages/thirdweb/src/wallets/wallet-types.ts:124]: Sure, can do tonight

**Review by gregfromstl — APPROVED** [review]

**Pull request merged**